src/gather.cpp

```diff
@@ -0,0 +1,89 @@
+/*******************************************************************************
+ *
+ * MIT License
+ *
+ * Copyright (c) 2024 Advanced Micro Devices, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ *******************************************************************************/
+
+#include <miopen/common.hpp>
+#include <miopen/find_solution.hpp>
+#include <miopen/gather.hpp>
+#include <miopen/gather/invoke_params.hpp>
+#include <miopen/gather/problem_description.hpp>
+#include <miopen/gather/solvers.hpp>
+#include <miopen/invoke_params.hpp>
+#include <miopen/logger.hpp>
+#include <miopen/miopen.h>
+#include <miopen/names.hpp>
+#include <miopen/tensor.hpp>
+
+namespace miopen {
+
```

Comment:
missing namespace `gather`?

Reply:
also add namespace gather to PR https://github.com/moreh-dev/MIOpen/pull/55/

Reply:
When implementing this operation, I referred to some previous general operations, such as [activation](https://github.com/moreh-dev/MIOpen/blob/develop/src/activ.cpp) or [softmax](https://github.com/moreh-dev/MIOpen/blob/develop/src/softmax.cpp), and noticed they don't declare specific namespaces (presumably for generality). Therefore, let's keep it as it is for now.

Reply:
updated.